src/groups/bmq/bmqu/bmqu_managedcallback.h

```diff
@@ -0,0 +1,270 @@
+// limitations under the License.
+
+// bmqu_managedcallback.h                                             -*-C++-*-
+#ifndef INCLUDED_BMQU_JSONPRINTER
+#define INCLUDED_BMQU_JSONPRINTER
+
+//@PURPOSE: Provide a mechanism to build functors in a reusable buffer.
+//
+//@CLASSES:
+//  bmqu::ManagedCallback: Mechanism to build functors in its buffer.
+//  bmqu::ManagedCallback::CallbackFunctor:
+//             Base interface for functors acceptable by bmqu::ManagedCallback.
+//
+//
+//@DESCRIPTION: 'bmqu::ManagedCallback' provides a mechanism to build functors
+// in its buffer.  The objects of this type are intended to be reused to avoid
+// buffer reallocations for better performance on critical paths.
+//
+/// Usage
+///-----
+// First, need to declare a functor class, and inherit it from the acceptable
+// callback type.
+//..
+//  struct ExampleCallback : public bmqu::ManagedCallback::CallbackFunctor {
+//      size_t& d_calls;
+//
+//      explicit ExampleCallback(size_t& calls)
+//      : d_calls(calls)
+//      {
+//          // NOTHING
+//      }
+//
+//      ~ExampleCallback() BSLS_KEYWORD_OVERRIDE
+//      {
+//          // NOTHING
+//      }
+//
+//      void operator()() const BSLS_KEYWORD_OVERRIDE { ++d_calls; }
+// };
+//..
+//
+// Next, you can use the defined callback type with bmqu::ManagedCallback:
+//..
+//  bmqu::ManagedCallback callback(allocator);
+//  size_t                counter = 0;
+//
+//  // A buffer will be reallocated here once to store the callback:
+//  callback.createInplace<ExampleCallback>(counter);
+//  for (int i = 0; i < 5; ++i) {
+//      callback();
+//  }
+//  // Has to reset `bmqu::ManagedCallback` object before reusing it:
+//  callback.reset();
+//
+//  for (int i = 0; i < 100; i++) {
+//      // There will be no buffer reallocations on these calls, because we
+//      // reuse the same `bmqu::ManagedCallback` object, and it has sufficient
+//      // buffer capacity to store `ExampleCallback`:
+//      callback.createInplace<ExampleCallback>(counter);
+//      callback();
+//      callback.reset();
+//  }
+//..
+//
+
+// BDE
+#include <bsl_functional.h>
+#include <bsl_utility.h>  // bsl::forward
+#include <bsl_vector.h>
+#include <bsls_assert.h>
+#include <bsls_compilerfeatures.h>
+
+#if BSLS_COMPILERFEATURES_SIMULATE_CPP11_FEATURES
+// Include version that can be compiled with C++03
+// Generated on Thu Feb 20 15:21:39 2025
+// Command line: sim_cpp11_features.pl bmqu_managedcallback.h
+#define COMPILING_BMQU_MANAGEDCALLBACK_H
+#include <bmqu_managedcallback_cpp03.h>
+#undef COMPILING_BMQU_MANAGEDCALLBACK_H
+#else
+
+namespace BloombergLP {
+namespace bmqu {
+
+// ===============
+// ManagedCallback
+// ===============
+
+class ManagedCallback BSLS_KEYWORD_FINAL {
+    /// The class useful for in-place construction and passing of functors
+    /// between different actors.
+  private:
+    // DATA
+    /// Reusable buffer holding the stored callback.
+    bsl::vector<char> d_callbackBuffer;
```

Comment:
Note that if the object stored in this buffer is not trivially copyable, we cannot simply copy `ManagedCallback` objects. For this reason I removed copy/assignment operators. There is an alternative to store this `d_callbackBuffer` under a shared_ptr, but I don't want to introduce this overhead unless we have a good reason to